test/helpers/storageQueries.ts

```diff
@@ -96,3 +96,111 @@ export async function processAllStorage(
 
   await limiter.disconnect();
 }
+
+export async function processRandomStoragePrefixes(
+  api: ApiPromise,
+  storagePrefix: string,
+  blockHash: string,
+  processor: (batchResult: { key: `0x${string}`; value: string }[]) => void,
+  override: string = ""
+) {
+  const maxKeys = 1000;
+  let total = 0;
+  const preFilteredPrefixes = splitPrefix(storagePrefix);
+  const chanceToSample = 0.05;
+  let prefixes = override
+    ? [override]
+    : preFilteredPrefixes.filter(() => Math.random() < chanceToSample);
+  if (prefixes.length > 25) {
+    prefixes = prefixes.slice(0, 25);
+  }
+  console.log(`Processing ${prefixes.length} prefixes: ${prefixes.join(", ")}`);
+  const limiter = rateLimiter();
+  const stopReport = startReport(() => total);
+
+  try {
+    await Promise.all(
+      prefixes.map(async (prefix) =>
+        limiter.schedule(async () => {
+          let startKey: string | undefined = undefined;
+          loop: for (;;) {
```

Comment:
We should not need a `goto` here.

Reply:
Replaced with a regular `while(true)` loop for easier reading. I had inherited it from the original function.